src/lib/LED/LED.h

```diff
@@ -23,104 +23,143 @@ void WS281BsetLED(uint32_t color)
+    if (esp_reset_reason() == ESP_RST_SW) {
+        state = NORMAL;
+    }
+    #endif
+    blinkyUpdateTime = 0;
+    hue = 0;
+    lightness = 128;
 }
-#endif
 
 void updateLEDs(uint32_t now, connectionState_e connectionState, uint8_t rate, uint32_t power)
 {
-#if (defined(PLATFORM_ESP32) && defined(GPIO_PIN_LED)) || defined(WS2812_LED_IS_USED)
-    constexpr uint32_t rate_colors[RATE_MAX] =
+    if (state == STARTUP)
     {
-        0x0000FF,     // 500/200 hz  blue
-        0x00FF00,     // 250/100 hz  green
-        0xFF8000,     // 150/50 hz   orange
-        0xFF0000      // 50/25 hz    red
+        blinkyUpdate(now);
+        return;
+    }
+
+    constexpr uint8_t rate_hue[RATE_MAX] =
+    {
+        170,     // 500/200 hz  blue
+        85,     // 250/100 hz  green
+        21,     // 150/50 hz   orange
+        0      // 50/25 hz    red
     };
-    constexpr uint32_t LEDupdateInterval = 100;
-    static uint8_t LEDfade = 0;
-    static int8_t LEDfadeAmount = 2;
-    static uint32_t LEDupdateCounterMillis;
+    constexpr uint32_t LEDupdateInterval = 50;
+    static uint32_t nextUpdateTime = 0;
 
     static connectionState_e lastState = disconnected;
-    static uint32_t lastColor = 0xFFFFFFFF;
+    static uint8_t lastRate = 0xFF;
     static uint32_t lastPower = 0xFFFFFFFF;
 
-    uint32_t color = rate_colors[rate];
-    if ((connectionState == disconnected) && (now - LEDupdateCounterMillis > LEDupdateInterval))
+    if ((connectionState == disconnected) && (now > nextUpdateTime))
     {
-        if (LEDfade == 30)
+        static uint8_t lightness = 0;
+        static int8_t dir = 1;
+
+        if (lightness == 64)
         {
-            LEDfadeAmount = -2;
+            dir = -1;
         }
-        else if (LEDfade == 0)
+        else if (lightness == 0)
         {
-            LEDfadeAmount = 2;
+            dir = 1;
         }
 
-        LEDfade += LEDfadeAmount;
-        LEDupdateCounterMillis = now;
+        lightness += dir;
+        nextUpdateTime = now + LEDupdateInterval;
         lastState = connectionState;
-        WS281BsetLED(
-            (color >> 16) * LEDfade / 255,
-            ((color >> 8) & 0xFF) * LEDfade / 255,
-            (color & 0xFF) * LEDfade / 255
-        );
+        WS281BsetLED(HsvToRgb(rate_hue[rate], 255, lightness));
     }
-    if (((connectionState != disconnected) && (lastState == disconnected)) || lastColor != color || lastPower != power)
+    if (((connectionState != disconnected) && (lastState == disconnected)) || lastRate != rate || lastPower != power)
     {
         lastState = connectionState;
-        lastColor = color;
+        lastRate = rate;
         lastPower = power;
-        int dim = fmap(power, 0, PWR_COUNT-1, 10, 255);
-        WS281BsetLED(
-            (color >> 16) * dim / 255,
-            ((color >> 8) & 0xFF) * dim / 255,
-            (color & 0xFF) * dim / 255
-        );
+        uint8_t lightness = fmap(power, 0, PWR_COUNT-1, 10, 128);
```

Comment:
I'm honestly not sure why we have fmap at all, but normal code should use `map()` which is less code. I know you didn't put the fmap here but if you're changing it, let's use the smaller version.

Reply:
`fmap` is a macro in crsf_protocol.h and actually smaller than `map`! Possibly because of the call overhead?